src/oc_erchef/apps/chef_objects/src/chef_cbv_cache.erl

```diff
@@ -0,0 +1,253 @@
+%% and tracks whether its mailbox has grown to large. If that happens, response with 'busy' instead
+%% of sending the request through.
+send_if_available(Msg) ->
+    case breaker_tripped() of
+        true -> busy;
+        false -> gen_server:call(?SERVER, Msg)
+    end.
+
+%% @doc intended for testing.
+stop() ->
+    gen_server:call(?SERVER, stop).
+
+% gen_server implementation
+
+init([Enabled, TTL]) ->
+    process_flag(trap_exit, true),
+    spawn_breaker(),
+    {ok, #state{enabled = Enabled,
+                ttl = round(TTL/2), % Splay is TTL/2 + (0-50% of TTL) for max of TTL.
+                claims = dict:new(),
+                tid = ets:new(chef_cbv_cache, [set, private, {read_concurrency, true}])
+               }}.
+handle_call(stop, _From, State) ->
+    exit(whereis(cbv_cache_breaker), kill),
+    {stop, normal, ok, State};
+handle_call(_Msg, _From, #state{enabled = false} = State) ->
+    {reply, undefined, State};
+handle_call({get, Key}, _From, #state{tid = Tid, claims = Claims} = State) ->
+    case dict:take(Key, Claims) of
+        error ->
+            % Nobody is currently working on it, so it's either here or not.
+            case ets:lookup(Tid, Key) of
+                [{_, Value}] -> {reply, Value, State};
+                [] -> {reply, undefined, State}
+            end;
+        {Pid, Claims1} ->
+            case is_process_alive(Pid) of
+                true -> {reply, retry, State};
+                false ->
+                    % If the original claiming process has died, it can't complete its
+                    % claim.
+                    case ets:lookup(Tid, Key) of
+                        [{_, Value}] -> {reply, Value, State};
+                        [] -> {reply, undefined, State#state{claims = Claims1}}
+                    end
+            end
+    end;
+handle_call({claim, Key}, {From, _Tag}, #state{claims = Claims} = State) ->
+    case dict:take(Key, Claims) of
+        error -> % Nobody has claimed it yet
+            {reply, ok, State#state{claims = dict:store(Key, From, Claims)}};
+        {From, _Claims1} ->
+            % This caller has already claimed it, that's fine..
+            {reply, ok, State};
+        {_OtherPid, _Claims1} -> % Someone has already claimed it, so caller should retry the GET
+            {reply, retry, State}
+    end;
+handle_call({put, Key, Value}, { From, _Tag }, #state{tid = Tid, ttl = TTL, claims = Claims} = State) ->
+    case dict:take(Key, Claims) of
+        error ->
+            % Mis-usage. Claim must be called before put.
+            {reply, {error, no_claim}, State};
+        { From, Claims1 } ->
+            % It is claimed and by this caller, they can go ahead and put.
+            % Update claims dictionary in state to remove the claim on this key
+            insert_into_cache(Tid, Key, Value, TTL),
+            {reply, ok, State#state{claims = Claims1}};
+        { _OtherPid, _Claims1 } ->
```

Comment:
This caluse doesn't get invoked as 163 matches always.I read the comment but is it still required ?